pkg/kubelet/config/apiserver.go

```diff
@@ -23,13 +23,27 @@ import (
 	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
 	"k8s.io/kubernetes/pkg/fields"
 	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
+	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/watch"
 )
 
 // NewSourceApiserver creates a config source that watches and pulls from the apiserver.
 func NewSourceApiserver(c *clientset.Clientset, nodeName types.NodeName, updates chan<- interface{}) {
-	lw := cache.NewListWatchFromClient(c.Core().RESTClient(), "pods", api.NamespaceAll, fields.OneTermEqualSelector(api.PodHostField, string(nodeName)))
```

Comment:
> Since the resource type's clientset interface already provides the same functionality. Why would we have two implementations doing the same thing?

I kind of like the one liner.  A one liner like that which took a `ClientPool` would be great.  Or maybe a generated function to return a listwatch.  @caesarxuchao